pkg/meta/metabuild/context_test.go

```diff
@@ -0,0 +1,200 @@
+			},
+			testVals: []any{true, false},
+		},
+		{
+			name: "clusteredIndexDefMode",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.GetClusteredIndexDefMode()
+			},
+			checkDefault: defVars.EnableClusteredIndex,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithClusteredIndexDefMode(val.(variable.ClusteredIndexDefMode))
+			},
+			testVals: []any{variable.ClusteredIndexDefModeOn, variable.ClusteredIndexDefModeOff},
+		},
+		{
+			name: "shardRowIDBits",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.GetShardRowIDBits()
+			},
+			checkDefault: defVars.ShardRowIDBits,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithShardRowIDBits(val.(uint64))
+			},
+			testVals: []any{uint64(6), uint64(8)},
+		},
+		{
+			name: "preSplitRegions",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.GetPreSplitRegions()
+			},
+			checkDefault: defVars.PreSplitRegions,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithPreSplitRegions(val.(uint64))
+			},
+			testVals: []any{uint64(123), uint64(456)},
+		},
+		{
+			name: "enableTablePartitionMode",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.GetEnableTablePartitionMode()
+			},
+			checkDefault: variable.On,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithEnableTablePartitionMode(val.(string))
+			},
+		},
+		{
+			name: "enableListTablePartition",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.EnableListTablePartition()
+			},
+			checkDefault: true,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithEnableListTablePartition(val.(bool))
+			},
+			testVals: []any{true, false},
+		},
+		{
+			name: "enableGlobalIndex",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.EnableGlobalIndex()
+			},
+			checkDefault: defVars.EnableGlobalIndex,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithEnableGlobalIndex(val.(bool))
+			},
+			testVals: []any{true, false},
+		},
+		{
+			name: "suppressTooLongIndexErr",
+			getter: func(ctx *metabuild.Context) any {
+				return ctx.SuppressTooLongIndexErr()
+			},
+			checkDefault: false,
+			option: func(val any) metabuild.Option {
+				return metabuild.WithSuppressTooLongIndexErr(val.(bool))
+			},
+			testVals: []any{true, false},
+		},
+		{
+			name: "is",
+			getter: func(ctx *metabuild.Context) any {
+				is, ok := ctx.GetInfoSchema()
+				require.Equal(t, ok, is != nil)
+				return is
+			},
+			checkDefault: nil,
+			option: func(val any) metabuild.Option {
+				if val == nil {
+					return metabuild.WithInfoSchema(nil)
+				}
+				return metabuild.WithInfoSchema(val.(infoschemactx.MetaOnlyInfoSchema))
+			},
+			testVals: []any{infoschema.MockInfoSchema(nil), nil},
+		},
+	}
+	defCtx := metabuild.NewContext()
+	allFields := make([]string, 0, len(fields))
+	for _, field := range fields {
+		switch val := field.checkDefault.(type) {
```

Comment:
maybe wrap inside `t.Run(...)`, easier to identify which case failed and we can run the subcase directly